### ISSUE TITLE
Options for image compression/optimization

### ISSUE THREAD

**Issue by CodeTheInternet** [bdkjones/CodeKit2#527]
Provide the ability to select different types of image compression, including "lossy" a la TinyPNG.

**Comment by KoltonG** [comment]
+1 for this, TinyPNG and TinyJPG compresses the file from 2MB to 325KB where CodeKit could not compress it by anything. I love CodeKit but I find myself still going to TinyPNG/JPG to compress my assets.

**Comment by blinkcursor** [comment]
I don't necessarily need options, just results. 

I've only recently started using Codekit in earnest (it rocks!) but the image compression results are not what I'd hoped for. The resulting files didn't look particularly small, so I just ran a side by side check with a smallish (800 x 600) jpg file. Optimised by codekit -> 583kb. I ran it through JPEGMini lossless compression -> 168kb. Much more in line with what I would expect.

**Comment by bdkjones** [comment]
Jpegmini is NOT lossless. The way it gets that much compression is by discarding data. This is not a fair comparison. Additionally, the jpegmini guys will not license their algorithm for use in apps. I've asked.

**Comment by blinkcursor** [comment]
Fair enough. Their website says no loss in quality which is perhaps not the same thing. Shame about the licensing attitude. 

Thanks for responding so promptly.

**Comment by blinkcursor** [comment]
Don't know if this is any help, but smashing magazine just published a really in-depth article about batch processing images with open source ImageMagick including a battery of tests to determine what are the optimum settings to use for best results. They are not lossless but judged by testing to be near indistinguishable.

http://www.smashingmagazine.com/2015/06/25/efficient-image-resizing-with-imagemagick/

**Comment by bdkjones** [comment]
Actually, this is a lot of help. Thanks!
